exercises/bowling/canonical-data.json

```diff
@@ -224,5 +224,21 @@
       "previousRolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 7, 3]
     },
     "expected": {"error": "Score cannot be taken until the end of the game"}
+  }, {
+    "description": "cannot roll after bonus roll for spare",
+    "property": "roll",
+    "input": {
+      "previousRolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 7, 3, 2],
```

Comment:
Maybe I am wrong here.  But the 2nd ball in the 10th is a spare, which rewards the roller with 1 bonus roll, 2, yielding a score of 12.

Reply:
You read it correctly. However, perhaps you missed the `roll` input field? In case there is any confusion, try pairing the rolls into frames.

| Frame No | Ball 1 | Ball 2 |
| --- | --- | --- |
| 1 | 0 | 0 |
| 2 | 0 | 0 |
| 3 | 0 | 0 |
| 4 | 0 | 0 |
| 5 | 0 | 0 |
| 6 | 0 | 0 |
| 7 | 0 | 0 |
| 8 | 0 | 0 |
| 9 | 0 | 0 |
| 10 | 7 | 3 |
| Bonus | 2 (valid) | 2 (from `"input": { "roll": 2 }`) |